Makefile.am

```diff
@@ -8,23 +8,23 @@ DIST_SUBDIRS = @TOP_DIST_SUBDIRS@
 
 $(GT_TARGETS_CLEAN):
 	@if [ -f $($@_stamp) ]; then \
-		export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) clean ; \
+		export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) clean || exit 1 ; \
 	fi
 	@rm -f $($@_stamp)
 $(GT_ALL_TARGETS_INSTALL):
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) install
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) install || exit 1
 $(GT_TARGETS_CHECK):
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) check
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) check || exit 1
 $(GT_ALL_TARGETS_DIST):
-	@$(MKDIR_P) package-output
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && MAKEFLAGS= $(MAKE) $(AM_MAKEFLAGS) dist
-	@cp $($@_SUBDIR)/$($@-tarball) package-output
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && MAKEFLAGS= $(MAKE) $(AM_MAKEFLAGS) dist && \
+	@$(MKDIR_P) package-output && \
+	@cp $($@_SUBDIR)/$($@-tarball) package-output || exit 1
```

Comment:
@msalle:
Comparing the original code and the modification, `mkdir` was originally done in another dir than `$($@_SUBDIR)` (assuming that `$($@_SUBDIR)` != `$PWD`), but with the modification it's invoked in `$($@_SUBDIR)` - unless that `MAKEFLAGS= $(MAKE) $(AM_MAKEFLAGS) dist` changes the `$PWD`.

Reply:
@msalle 
I retriggered the ["deploy" job](https://travis-ci.org/gridcf/gct/jobs/642843741) and it errors out the same way as during the original run, hence I tend to assume this error could be related to the changes and not an intermittent issue with Travis CI:

```
[...]
make[1]: Leaving directory `/gct/common/source'

/bin/sh: line 1: @/usr/bin/mkdir: No such file or directory

make: *** [globus_common-dist] Error 1

+ rm -f /tmp/tmp.FkzDiIlIxc

The command "travis-ci/setup_tasks.sh" exited with 2.
```

OTOH, when checking on my local Ubuntu 19.04 `mkdir` is not in `/usr/bin` but still in `/bin` - i.e. `/bin` and `/sbin` aren't yet "merged" with `/usr/bin` and `/usr/sbin` as on Fedora. Not sure why this should break now, though. :-/

Reply:
You need to drop the leading `@` from the new `$(MKDIR_P)` line and the `cp` line below it since they're now part of the same shell invocation (due to the `&& \`) at the end of the previous lines. The error is not that it's not finding `mkdir` in `/usr/bin`, the error is that it's looking for a file named `@/usr/bin/mkdir`.

Reply:
Good point, thanks. It should be fixed now.
We should at some point further cleanup the Makefile.

Reply:
@matyasselmeci 
> You need to drop the leading `@` from the new `$(MKDIR_P)` line and the `cp` line below it since they're now part of the same shell invocation (due to the `&& \`) at the end of the previous lines. The error is not that it's not finding `mkdir` in `/usr/bin`, the error is that it's looking for a file named `@/usr/bin/mkdir`.

Thanks for the clarification. Actually that even was my initial idea: that some "unexpanded" command is fed to the shell. But then I thought why should the `@` make keyword loose meaning in a command list? But yeah, the manual tells (from https://www.gnu.org/software/make/manual/make.html#Recipe-Echoing):

```
[...]
When a line starts with ‘@’, the echoing of that line is suppressed.
The ‘@’ is discarded before the line is passed to the shell. [...]
```
...so it's only discarded at the start of the line. I'd then actually write it like that instead:

```
@ command arg1 arg2 [...]
```

...then it could be understandable from the context that `@` does not affect the command, but the whole line - but I could also RTFM instead, of course. ;-)